R/local.R

```diff
@@ -133,8 +133,13 @@ local_reproducible_output <- function(width = 80,
     .local_envir = .env
   )
 
-  if (isTRUE(capabilities("NLS")) && Sys.getenv("LANG") != "C") {
-    withr::local_language(lang, .local_envir = .env)
+  if (isTRUE(capabilities("NLS"))) {
+    # FIXME: We need a new `locale` argument here, should be consistent
+    # with the `lang` argument to `local_reproducible_output()`
+    withr::local_envvar(c(LC_ALL = "en_GB"), .local_envir = .env)
+    if (!(Sys.getenv("LANG") %in% c("C", "C.UTF-8"))) {
+      withr::local_language(lang, .local_envir = .env)
+    }
```

Comment:
How about setting unconditionally? And document in `@param lang` that callers might need to set `LC_ALL` in some way, but that it's tricky to do it in a cross-platform way?

```suggestion
    suppressWarnings(
      withr::local_language(lang, .local_envir = .env)
    )
```